### PR TITLE
upkeep: add TS support for `useTaxonomy` hook

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

Closes #

### Changelog Entry

> Changed - add TS support for `useTaxonomy` hook

### Credits

Props @fabiankaegy @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
     "@emotion/styled": "^11.10.5",
     "@floating-ui/react-dom": "^1.0.1",
     "@leeoniya/ufuzzy": "^1.0.8",
+    "@wordpress/core-data": "^6.29.0",
```
Comment (Sidsector9) [inline package.json]: @fabiankaegy I've installed this so that I can import the `Taxonomy` type. Can you confirm if this is the correct way to go?

Comment (fabiankaegy) [reply package.json]: @Sidsector9 I would move it to a `devDependency` because the package won't actually get bundled. But yeah this makes sense to me :)

**Comment by fabiankaegy** [comment]
@Sidsector9 I'm really surprised by the amount of changes in the package lock file. Are you by chance using a different NPM version?

**Comment by Sidsector9** [comment]
@fabiankaegy I'm on NPM v8.19.4.
I just tried fixing it, can you please check?

**Review by fabiankaegy — CHANGES_REQUESTED** [review]
It seems Git didn't pick up that you renamed the file. The old .js version still is there and is getting used

**Review by fabiankaegy — APPROVED** [review]

**Pull request merged**